pkg/apis/azure/types_infrastructure.go

```diff
@@ -167,7 +167,13 @@ type Subnet struct {
 	Zone *string
 	// Migrated is set when the network layout is migrated from NetworkLayoutSingleSubnet to NetworkLayoutMultipleSubnet.
 	// Only the subnet that was used prior to the migration should have this attribute set.
-	Migrated bool
+	Migrated       bool
+	NatGatewayName *string
+}
+
+// HasNatGateway returns if NAT is enabled
+func (s Subnet) HasNatGateway() bool {
+	return s.NatGatewayName != nil
```

Comment:
I don't see this function getting used.

Reply:
it's used in the 2nd PR for helm chart installation